client/app/src/main/java/net/tbp/interval/ui/backup/Signup.java

```diff
@@ -0,0 +1,140 @@
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+
+import net.tbp.interval.ui.reminder.ReminderProvider;
+
+public class Signup extends Activity {
+    private Button signinBtn;           // button for user to go sign up
+    private Button submitBtn;           // button for user to go submit
+    private FirebaseAuth mAuth;
+    private EditText email, password, confirmPassword;
+    private TextView signupStatus;
+    String FIREBASETAG = "Firebase";
+    protected void onCreate(Bundle savedInstanceState) {
+        // set theme for screen
+        setTheme(R.style.blackBackground);
+        // create the screen
+        super.onCreate(savedInstanceState);
+        // load content from add_reminder fragment to render the screen
+        setContentView(R.layout.sign_up);
+
+        // Initialize Firebase Auth
+        mAuth = FirebaseAuth.getInstance();
+
+        email = findViewById(R.id.emailSignup);
+        password = findViewById(R.id.passwordSignup);
+        confirmPassword = findViewById(R.id.confirmPasswordSignup);
+        signupStatus = findViewById(R.id.signupStatus);
+
+        //add func to signin button
+        signinBtn = (Button) findViewById(R.id.gotoSigninBtn);
+        signinBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                // initial intent that will use to call Signin class to let user signin their info
+                Intent informationIntent = new Intent(v.getContext(), Signin.class);
+                startActivity(informationIntent);
+                finish();
+            }
+        });
+
+        //add func to submit button
+        submitBtn = (Button) findViewById(R.id.submitSignupBtn);
+        submitBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                // to add sqlite
+                ContentValues values = new ContentValues();
+                signupStatus.setText("");
+                String userEmail = email.getText().toString().trim();
+                String userPassword1 = password.getText().toString().trim();
+                String userPassword2 = confirmPassword.getText().toString().trim();
+                //check condition
+                if(TextUtils.isEmpty(userEmail)){
+                    email.setError("Email is required.");
+                    return;
+                }
+                if(TextUtils.isEmpty(userPassword1)){
+                    password.setError("Password is required.");
+                    return;
+                }
+                if(TextUtils.isEmpty(userPassword2)){
+                    confirmPassword.setError("Confirm password is required.");
+                    return;
+                }
+                if(userPassword1.length() < 8){
```

Comment:
should communicate to the user that the password must be 8 characters or longer

Reply:
since the above logic checks that password1 and password2 need to be equal. So we will just need to check only one length of password